uvicorn/protocols/utils.py

```diff
@@ -20,9 +15,17 @@ def get_remote_addr(transport):
         else:
             family = socket_info.family
 
-        if family in SUPPORTED_SOCKET_FAMILIES:
+        if family in (socket.AF_INET, socket.AF_INET6):
             return (str(info[0]), int(info[1]))
-
+        elif hasattr(socket, "AF_UNIX") and family is socket.AF_UNIX:
+            if isinstance(info, tuple):
+                # fd case
+                # <uvloop.PseudoSocket fd=13, family=AddressFamily.AF_UNIX, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 38634)>
+                return (str(info[0]), int(info[1]))
+            else:
+                # unix socket case
+                # <uvloop.PseudoSocket fd=21, family=AddressFamily.AF_UNIX, type=SocketKind.SOCK_STREAM, proto=0, laddr=/tmp/gunicorn.sock>
+                return None
```

Comment:
I think I'd do something like this:

```python
HAS_AF_UNIX = hasattr(socket, "AF_UNIX")

def get_remote_addr(transport):
    ...

        if family in (socket.AF_INET, socket.AF_INET6) or (
            HAS_AF_UNIX and family == socket.AF_UNIX and isinstance(family, tuple)
        ):
            return (str(info[0]), int(info[1]))

        return None
```

Reply:
Could we just change the whole block to...

```python
return (str(info[0]), int(info[1])) if isinstance(info, tuple) else None
```

Reply:
I did it that way, way simpler,
Just note that I kept in `get_remote_addr` the try except introduced it seems in #495